File: app/src/main/java/beans/prj/newsapp/di/NewsFragBuilder.java
package beans.prj.newsapp.di;

import beans.prj.newsapp.ui.home.NewsListFragment;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class NewsFragBuilder {

    @ContributesAndroidInjector
    abstract NewsListFragment bindNewsListFragment();
}

File: app/src/main/java/beans/prj/newsapp/di/NewsFragBuilder.kt
package beans.prj.newsapp.di

import beans.prj.newsapp.ui.home.NewsListFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class NewsFragBuilder {
    @ContributesAndroidInjector
    abstract fun bindNewsListFragment(): NewsListFragment?
}